src/components/AddHouseButton.vue

```diff
@@ -0,0 +1,337 @@
+            :class="{
+              'p-invalid': v$.houseData.address.zip_code.$error,
+            }"
+            @blur="v$.houseData.address.zip_code.$touch"
+          />
+          <small v-if="v$.houseData.address.zip_code.$error" class="p-error">{{
+            v$.houseData.address.zip_code.$errors[0].$message
+          }}</small>
+        </p>
+      </div>
+    </form>
+    <template #footer>
+      <Button
+        label="Зберегти зміни"
+        icon="pi pi-check"
+        @click="addNewHouse"
+        autofocus
+        class="p-button-info"
+        type="button"
+        value="Submit"
+      />
+      <Button
+        label="Скасувати зміни"
+        icon="pi pi-times"
+        @click="closeAddHouseModal"
+        class="p-button-outlined p-button-info"
+      />
+    </template>
+  </Dialog>
+</template>
+
+<script lang="ts">
+import { defineComponent } from 'vue';
+import useVuelidate from '@vuelidate/core';
+import {
+  requiredValidator,
+  flatQuantityValidator,
+  houseAreaValidator,
+  adjoiningAreaValidator,
+  regionValidator,
+  cityValidator,
+  districtValidator,
+  streetValidator,
+  houseBlockNumberValidator,
+  houseNumberValidator,
+  zipCodeValidator,
+} from '@/utils/validators';
+import Button from 'primevue/button';
+import Dialog from 'primevue/dialog';
+import InputText from 'primevue/inputtext';
+import { StoreModuleEnum } from '@/store/types';
+import { HousesActionsEnum } from '@/store/houses/types';
+
+export default defineComponent({
+  name: 'AddHouse',
+  components: {
+    Dialog,
+    Button,
+    InputText,
+  },
+  props: {
+    id: {
+      type: Number,
+      required: true,
+    },
+  },
+  data() {
+    return {
+      houseData: {
+        quantity_flat: '',
+        house_area: '',
+        adjoining_area: '',
+        address: {
+          region: '',
+          city: '',
+          district: '',
+          street: '',
+          house_block: '',
+          house_number: '',
+          zip_code: '',
+        },
+      },
+      v$: useVuelidate(),
+    };
+  },
+  validations() {
+    return {
+      houseData: {
+        quantity_flat: { requiredValidator, flatQuantityValidator },
+        house_area: { requiredValidator, houseAreaValidator },
+        adjoining_area: { requiredValidator, adjoiningAreaValidator },
+        address: {
+          region: { requiredValidator, regionValidator },
+          city: { requiredValidator, cityValidator },
+          district: { requiredValidator, districtValidator },
+          street: { requiredValidator, streetValidator },
+          house_block: { requiredValidator, houseBlockNumberValidator },
+          house_number: { requiredValidator, houseNumberValidator },
+          zip_code: { requiredValidator, zipCodeValidator },
+        },
+      },
+    };
+  },
+  methods: {
+    openAddHouseModal() {
+      this.$store.dispatch(`${StoreModuleEnum.housesStore}/${HousesActionsEnum.SET_ADD_HOUSE_MODAL}`, true);
```

Comment:
Do we use displayAddHouseModal somewhere except this component? Maybe it would be better to move this field from store to component.

Reply:
No, we don't) Got it, already fixed